### PR TITLE
BioFormats : Matlab unit tests 

### PR DESCRIPTION
**Pull request by bramalingam**
Majorly check's Octave compatibility and a few minor additions.
Testing instructions in Matlab:

1) Download bfMatlab and add it to the Matlab path.
ex: ">> addpath('/Users/bramalingam/Downloads/bfmatlab-5')" 

2) Download the matlab-xunit framework from
https://github.com/psexton/matlab-xunit
ex: ">> addpath(genpath('/Users/bramalingam/OME/matlab-xunit/src'))"

3) Now change the working directory of matlab to the following directory (in your bioformats repository),
ex: ">> cd('/Users/bramalingam/OME/bioformats/components/formats-gpl/test/matlab')"

4) Run the package : 
">> runxunit ./"

### CODE REVIEW

File: components/formats-gpl/test/matlab/TestBfCheckJavaPath.m [hunk 1]
```diff
@@ -78,5 +80,11 @@ function testPerformance(self)
             % bfCheckJavaPath should return fast
             assertTrue(mean(times(3:end)) < self.maxTime);
         end
+        
+        function testJavaMethod(self)
+            self.version = char(loci.formats.FormatTools.VERSION);
+            version = char(java_get('loci.formats.FormatTools', 'VERSION'));
+            assertEqual( self.version, version);
```
Comment (sbesson) [inline components/formats-gpl/test/matlab/TestBfCheckJavaPath.m]: Can we also test the signature of `bfCheckJavaPath()` where the version is returned as a second output is returned here, i.e.

```
[self.status, self.version] = bfCheckJavaPath(false);
```

File: components/formats-gpl/test/matlab/TestBfGetFileExtensions.m [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        
+        function self = TestBfGetFileExtensions(name)
+            self = self@TestBfMatlab(name);
+        end
+        
+        function testOctaveComaptibility(self)
+            
+            self.readers = javaMethod('getReaders', javaObject('loci.formats.ImageReader'));
+            
+            fileExt = cell(numel(self.readers), 2);
```
Comment (sbesson) [inline components/formats-gpl/test/matlab/TestBfGetFileExtensions.m]: Not 100% on how this test scales? It seems the logic of `bfGetFileExtensions` is reimplemented in the test but the function itself is never called.

Comment (bramalingam) [inline components/formats-gpl/test/matlab/TestBfGetFileExtensions.m]: The external PR had reimplemented our Matlab implementation of arrarfun, to fit Octave compatibility. Was just checking the same over here. Thinking about it, this is more or less testing Matlab functions and not our methods..! May be not worth the addition.

File: components/formats-gpl/test/matlab/TestBfGetFileExtensions.m [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            for i = 1:numel(self.readers)
+                suffixes = self.readers(i).getSuffixes();
+                fileExt{i, 1} = arrayfun(@char, suffixes, 'Unif', false);
```
Comment (bramalingam) [inline components/formats-gpl/test/matlab/TestBfGetFileExtensions.m]: Line 47 should have been deleted.

File: components/formats-gpl/test/matlab/TestBfopen.m [hunk 4]
```diff
@@ -30,6 +31,13 @@
         filepath
         data
         nSeries = 1
+        flag = 1
+        width = 10;
+        height = 10;
+        x = 10;
+        y = 10;
+        DefaultSize = [512,512];
```
Comment (sbesson) [inline components/formats-gpl/test/matlab/TestBfopen.m]: This should already be given by `sizeX` / `sizeY` properties initialized in `ReaderTest`

**Comment by joshmoore** [comment]
Note: this also can't be automatically merged.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#971](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/971/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/971/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - components/formats-gpl/test/matlab/TestBfopen.m

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#973](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/973/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/973/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - components/formats-gpl/test/matlab/TestBfopen.m

**Comment by sbesson** [comment]
@bramalingam: can you merge `origin/develop` into that branch?

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#975](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/975/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/975/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - components/formats-gpl/test/matlab/TestBfopen.m

**Comment by bramalingam** [comment]
@sbesson : Pushed all changes as requested. Can you please remove the Exclude label?

**Comment by melissalinkert** [comment]
Label removed.

**Comment by joshmoore** [comment]
From https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-5.1-merge-matlab/754/ (now at 12 hours rather than the average of <3 min):
```
[Warning: *** Insufficient memory detected. ***
*** 469m found ***
*** 569m or greater is recommended ***
*** See http://www.mathworks.com/matlabcentral/answers/92813 ***
*** for instructions on increasing memory allocation. ***
] 
[> In bfCheckJavaMemory at 53
  In TestBfCheckJavaMemory>TestBfCheckJavaMemory.runJavaMemoryCheck at 43
  In TestBfCheckJavaMemory>TestBfCheckJavaMemory.testLowMemory at 62
  In TestCase>TestCase.run at 74
  In TestSuite>TestSuite.run at 85
  In TestSuite>TestSuite.run at 85
  In runxunit at 136] 
[Warning: Objects of loci/formats/UpgradeChecker class exist - not clearing java]
```

**Comment by joshmoore** [comment]
@bramalingam : perhaps the test of the updater (https://github.com/openmicroscopy/bioformats/pull/1781/files#diff-b18d3579d0aeb5c695e4a5f17ab13c15R44) or of the classpath itself has left something stuck in memory?

**Comment by bramalingam** [comment]
Need to do implement a teardown function I guess. Looking into it.

**Comment by bramalingam** [comment]
@joshmoore : The above commit's should fix the issues.

**Comment by sbesson** [comment]
Looks good to me. Validated by https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-matlab/759/'s last run. Barring any veto from @melissalinkert, I propose we merge this PR and include #1757 in tonight's build.

**Pull request merged**